### PR TITLE
infra/aws/tf/infrastructure-services: set up healthcheck for google-pacakges-apt

### PR DESCRIPTION
**Pull request by iftachk**
- introduce two new modules: 
  -  sns-via-email: setup sns topic and associated email subscription 
  - external-resource-health-check/https-health-check: setup HTTPS health checks per provided config

Requires a new backend s3 bucket per discussion with @ameukam 

ref: #4841 
/cc @xmudrii
/assign @ameukam

### CODE REVIEW

File: infra/aws/terraform/infrastructure-services/main.tf [hunk 1]
```diff
@@ -0,0 +1,61 @@
...
+  backend "s3" {
+    bucket = "<NEW_BUCKET_NAME>"
+    key    = "infrastructure-services/terraform.tfstate"
+    region = "us-east-2"
+  }
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 4.47"
```
Comment (ameukam) [inline infra/aws/terraform/infrastructure-services/main.tf:29]: Do we want to use the v5 version ?

File: infra/aws/terraform/modules/sns-via-email/main.tf [hunk 2]
```diff
@@ -0,0 +1,28 @@
+resource "aws_sns_topic" "this" {
+  name = var.name
+  delivery_policy = jsonencode({
```
Comment (ameukam) [inline infra/aws/terraform/modules/sns-via-email/main.tf:3]: This will enforce the policy for every SNS topic and don't give room for customization. Can use pass the path of a json file to the variable ?

File: infra/aws/terraform/modules/external-resource-health-check/https-health-check/variables.tf [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+  default = "30"
+}
+
+variable "regions" {
+  type = list(string)
+  description = "AWS regions - us-east-1 | us-west-1 | us-west-2 | eu-west-1 | ap-southeast-1 | ap-southeast-2 | ap-northeast-1 | sa-east-1"
+  default = [
+    "us-east-1",
+    "us-west-1",
+    " us-west-2",
```
Comment (ameukam) [inline infra/aws/terraform/modules/external-resource-health-check/https-health-check/variables.tf:35]:
```suggestion
    "us-west-2",
```

**Comment by xmudrii** [comment]
@iftachk @ameukam This looks interesting, but is this something that's actually helpful to us at this point? I have some concerns:
- who is monitoring the given email address?
- is it possible to receive notifications only via email? Can we eventually integrate with Slack?
- how are we going to act on alerts?
- how are we going to pass those notifications to users?

My biggest concern is that we don't have any control over that repository and we don't have a good way to pass notifications down to users, meaning that this is eventually useful only to us as a signal. Given that Google repositories are going to be phased out soon™, I think that we should focus on making a solution that's going to work with OBS. That's something that we should collaborate with OBS team and see what's the best way to accomplish something similar.

Regardless, I think this is a really nice idea, we should just see how can we shape it so it provides value to a broader group (including end users).

**Comment by iftachk** [comment]
@xmudrii 
Thank you for your feedback

- who is monitoring the given email address? 
That is a good question, I think the objective is to increase observability.

- is it possible to receive notifications only via email? Can we eventually integrate with Slack?
I can add another HTTPS endpoint SNS to utilize Slack's incoming webhook
Or we can [notify-slack](https://registry.terraform.io/modules/terraform-aws-modules/notify-slack/aws/latest) TF module or we use this 

- how are we going to act on alerts?
Another good question, the focus in my opinion is to have these alerts delivered to the people with access to remedy the issue, if they do not have any alerts set already (shrug)

- how are we going to pass those notifications to users?
Can you clarify what you mean by users? End users eg consumers / those that created issues of the APT GPG errors

Bottom line, responsibility needs to be established and agreed upon with all relevant stakeholders

**Comment by xmudrii** [comment]
> Another good question, the focus in my opinion is to have these alerts delivered to the people with access to remedy the issue, if they do not have any alerts set already (shrug)

The issue here is that we don't know who have access to the platform. There's no one from our community (not even Google Build Admins) that have access to remedy the issue. I think it's safe to assume that Google employees maintaining that platform have some sort of monitoring and alerting and this wouldn't be much of use to them.

This is going to significantly change with OBS, however. Part of OBS infra is maintained by the OBS team at SUSE, but another part is maintained by us (SIG K8s infra and SIG Release). Proper alerting will be very helpful to know if there's anything going on that needs our attention.

> Can you clarify what you mean by users? End users eg consumers / those that created issues of the APT GPG errors

Yes, consumers that are using those repos.
